### PR TITLE
Fixed changing background image based on light/dark theme

### LINKED ISSUE #8526: Poor contrast on Supercharge your apps with Red Hat banner graphic's text on Dashboard
### Bug description

I'm running Podman Desktop 1.12 on my Mac and the graphic at the bottom of the dashboard advertising the Red Hat extension pack uses white text on a light pink background, which is very hard to read. Could something be done next version to ensure there is sufficient contrast to easily read?

Screenshot 2024-08-20 at 12 27 42 PM

### Operating system

M2 Pro MacBook 14.6.1

### Installation Method

Installer from website/GitHub releases

### Version

1.12.0

### Steps to reproduce

Open Podman Desktop and look at the bottom of the Dashboard area to see the graphic

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gastoner**
### What does this PR do?
Fix changing background image based on light/dark theme

### Screenshot / video of UI

https://github.com/user-attachments/assets/eff29de6-1c78-4eb3-82b8-41de1482ddb4

### What issues does this PR fix or reference?
Close #8526 

### How to test this PR?
Switch between light/dark theme and check if the recommendation banners reflect the change

### CODE REVIEW

File: packages/renderer/src/lib/recommendation/ExtensionBanner.svelte [hunk 1]
```diff
@@ -13,7 +13,7 @@ let style: string | undefined = undefined;
 let hasBackground: boolean =
   !!banner.background && ((!!banner.background.dark && !!banner.background.light) || !!banner.background.gradient);
 
-onMount(() => {
+afterUpdate(() => {
```
Comment (axel7083) [inline packages/renderer/src/lib/recommendation/ExtensionBanner.svelte]: afterUpdate is deprecated in svelte 5 https://svelte-5-preview.vercel.app/docs/deprecations

Comment (benoitf) [reply packages/renderer/src/lib/recommendation/ExtensionBanner.svelte]: I think the update needs to occur when isDark is updated

**Comment by gastoner** [comment]
Updated file to svelte5

File: packages/renderer/src/lib/recommendation/ExtensionBanner.svelte (hunk 1 above)
Comment (gastoner) [reply packages/renderer/src/lib/recommendation/ExtensionBanner.svelte]: Yes I've pushed conversion to svelte5 which could also work (option 1)
Or just add `$: if (isDark) {
  changeStyle();
}` function which will be also called in `onMount()` (option 2)
@benoitf @axel7083

**Review by axel7083 — COMMENTED** [review]
https://github.com/user-attachments/assets/9fc67499-0043-463d-996c-8b8edff9c5e2

When trying to very quickly change the theme, I am able to have some opposite result

**Comment by gastoner** [comment]
> Screencast.from.2024-09-04.16-34-14.mp4
> 
> When trying to very quickly change the theme, I am able to have some opposite result

Hmm, I got the same result even with the second option that I've proposed. Any ideas how to make this quicker or fix this, because it seems like if you return to the dashboard when is the light theme same, the image wont change

**Comment by axel7083** [comment]
The `ExtensionBanners.svelte` fetch the isDark on mount, and it is not refresh as the component is reused.

https://github.com/containers/podman-desktop/blob/aba56f0e96f92bcecd9db6ff556b2dfd2b6b1341/packages/renderer/src/lib/recommendation/ExtensionBanners.svelte#L11-L15

**Comment by gastoner** [comment]
@axel7083  try it now!

**Comment by axel7083** [comment]
> The `ExtensionBanners.svelte` fetch the isDark on mount, and it is not refresh as the component is reused.
> 
> https://github.com/containers/podman-desktop/blob/aba56f0e96f92bcecd9db6ff556b2dfd2b6b1341/packages/renderer/src/lib/recommendation/ExtensionBanners.svelte#L11-L15

We might need to either, get the `isDark` at every render (possible but not really performance wise) or trying to get an event listener for it :?

**Comment by gastoner** [comment]
> > The `ExtensionBanners.svelte` fetch the isDark on mount, and it is not refresh as the component is reused.
> > https://github.com/containers/podman-desktop/blob/aba56f0e96f92bcecd9db6ff556b2dfd2b6b1341/packages/renderer/src/lib/recommendation/ExtensionBanners.svelte#L11-L15
> 
> We might need to either, get the `isDark` at every render (possible but not really performance wise) or trying to get an event listener for it :?

The listener would be probably good, now it should get the boolean value each time user enters the dashboard

**Comment by benoitf** [comment]
it looks like ApperanceUtil isDark or getTheme should be removed and it should be a state (v5) or store v4 being updated/refreshed when configuration is changing

**Comment by axel7083** [comment]
> it looks like ApperanceUtil isDark or getTheme should be removed and it should be a state (v5) or store v4 being updated/refreshed when configuration is changing

The ApperanceUtil is doing some _hacky_ stuff to determine the theme

https://github.com/containers/podman-desktop/blob/9c946fddb1c614e7c0db0ff4a59c50f33d5e8d06/packages/renderer/src/lib/appearance/appearance-util.ts#L41-L53

Looking at the api of MediaQueryList there is a `addEventListener` and `removeEventListener` which would allow the base system for a quick hack.

But since we are using a configuration value (light / dark / system), we might want to deprecate the AppearanceUtil and have a store as suggested by @benoitf

**Comment by gastoner** [comment]
> it looks like ApperanceUtil isDark or getTheme should be removed and it should be a state (v5) or store v4 being updated/refreshed when configuration is changing

Should I convert it to v5 then? I've done something like this https://github.com/containers/podman-desktop/commit/6db3c75bc2814a84aa31a9bb84db1187396130f0

**Comment by feloy** [comment]
The PR breaks the "system" appearance:
- When you switch to the system appearance, the latest selected one is kept (if you were in dark mode, selecting the system mode displays the dark mode, even if the system mode is light)
- changing the system mode does not change the Podman Desktop mode reactively (it was changing it before)

Before:
https://github.com/user-attachments/assets/66329798-22b8-41ad-84ea-1426c225950c

After:
https://github.com/user-attachments/assets/ceacccf5-de49-449c-8527-0bc316d8a0b4

**Review by feloy — CHANGES_REQUESTED** [review]
see comment above

**Comment by gastoner** [comment]
@feloy This PR needs to be updated after will be #8759 closed, probably should convert it to draft for now

**Comment by feloy** [comment]
> @feloy This PR needs to be updated after will be #8759 closed, probably should convert it to draft for now

sorry, my comment was for #8759

**Comment by gastoner** [comment]
There would be a need to improve system change detection (similar problem in MonacoEditor)

File: packages/renderer/src/lib/recommendation/ExtensionBanners.svelte [hunk 2]
```diff
@@ -1,10 +1,29 @@
...
+onDestroy(() => {
+  if (isDarkUnsubscribe) {
+    isDarkUnsubscribe();
+  }
+});
 </script>
 
 {#each $extensionBannerInfos as banner (banner.extensionId)}
-  <ExtensionBanner banner={banner} isDark={$isDark} />
+  <ExtensionBanner banner={banner} isDark={isDarkTheme} />
```
Comment (feloy) [inline packages/renderer/src/lib/recommendation/ExtensionBanners.svelte]: As far as I understand svelte reactivity, it seems to me that the changes in this file are not necessary, we are just adding a layer of reactivity.
(and testing without these changes, it works the same)

Comment (gastoner) [reply packages/renderer/src/lib/recommendation/ExtensionBanners.svelte]: I see, thank you, I thought that I've forgot to use subscribe here

**Review by feloy — APPROVED** [review]

**Review by benoitf — APPROVED** [review]
nice to see using also the new svelte5 syntax

**Pull request merged**